packages/modular-scripts/src/utils/getPackageDependencies.ts

```diff
@@ -69,26 +71,41 @@ export async function getPackageDependencies(
     targetManifest.dependencies,
   ) as Dependency;
 
-  const lockDeps = await buildDepTree(
-    // Build a dependency tree from the lockfile, using target dependencies and root dependencies in order of specificity
-    JSON.stringify(
-      Object.assign(Object.create(null), targetManifest, {
-        dependencies: Object.assign(
-          Object.create(null),
-          rootManifest.dependencies,
-          targetManifest.dependencies,
-        ) as Dependency,
-        devDependencies: Object.assign(
-          Object.create(null),
-          rootManifest.devDependencies,
-          targetManifest.devDependencies,
-        ) as Dependency,
-      }),
-    ),
-    lockFile,
-    true,
-    LockfileType.yarn,
-  );
+  let lockDeps: Dependency = {};
+
+  try {
+    // Try to parse with v1 parser
+    const parsedLockfile = lockfile.parse(lockFile);
+    lockDeps = Object.entries(deps).reduce<Record<string, string>>(
+      (acc, [name, version]) => {
+        acc[name] = (parsedLockfile.object as LockFileEntries)[
+          `${name}@${version}`
+        ].version;
+        return acc;
+      },
+      {},
+    );
+  } catch (e) {
+    // Try to parse as yaml (v2+) - https://github.com/yarnpkg/yarn/issues/5629#issuecomment-753418765
+    const parsedLockfile = yaml.load(lockFile) as LockFileEntries;
+    const dependencyArray = Object.entries(deps);
+    // Parse: yarn v3 lock comes with entries like "'yargs@npm:^15.0.2, yargs@npm:^15.1.0, yargs@npm:^15.3.1, yargs@npm:^15.4.1'"
+    Object.entries(parsedLockfile).forEach(([name, { version }]) => {
+      const entryDependencies = name.split(', ');
+      dependencyArray.some(([dependencyName, dependencyVersion]) => {
```

Comment:
As you're not actually using the return value of `Array.prototype.some`, a for loop with a break statement would be clearer.

Reply:
I find it quite idiomatic when we ignore the return value (functional breaking loop), but no prob, done.